core-backend/common/src/funcs.rs

```diff
@@ -48,8 +49,9 @@ pub struct FuncsHandler<Ext: Externalities + 'static, Runtime> {
 impl<Ext, R> FuncsHandler<Ext, R>
 where
     Ext: BackendExternalities + 'static,
-    Ext::Error: BackendExternalitiesError,
-    Ext::AllocError: BackendAllocExternalitiesError<ExtError = Ext::Error>,
+    Ext::InfallibleError: BackendExternalitiesError,
```

Comment:
I don't like `Infallible*` naming here. It's more `UnrecoverableError` that means that couldn't be caught in userspace